### PR TITLE
telemetry(messages): maybeShowMinVscodeWarning

### PR DESCRIPTION
**Pull request by justinmk3**
## Problem

no metrics for notifications.

## Solution

add `toolkit_showNotification` to `showMessage` and use it in `maybeShowMinVscodeWarning`.

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/shared/utilities/messages.ts [hunk 1]
```diff
@@ -35,21 +35,32 @@ export function makeFailedWriteMessage(filename: string): string {
...
-        case 'error':
-        default:
-            return vscode.window.showErrorMessage(message, { modal: useModal }, ...items)
+    options: vscode.MessageOptions & { telemetry?: boolean } = {},
+    metric: ToolkitShowNotification = {
+        id: 'unknown',
+        component: 'editor',
     }
+): Thenable<string | undefined> {
+    return telemetry.toolkit_showNotification.run(async (span) => {
```
Comment (nkomonen-amazon) [inline packages/core/src/shared/utilities/messages.ts:48]: nice idea

Comment (justinmk3) [reply packages/core/src/shared/utilities/messages.ts:48]: Yeah I need to think more about this, could you show a pseudocode example of what you have in mind? 

Meanwhile, I will push a lower-risk change for today's release.

Comment (justinmk3) [reply packages/core/src/shared/utilities/messages.ts:46]: Ah, I think I can make that work in a followup PR.

File: packages/core/src/shared/utilities/messages.ts [hunk 2]
```diff
@@ -75,7 +86,16 @@ export async function showMessageWithUrl(
...
 
-    const p = showMessageWithItems(message, kind, items, useModal)
+    const p = showMessage(
+        kind,
+        message,
+        items,
+        { modal: useModal },
+        {
+            id: 'showMessageWithUrl',
+            reasonDesc: getTelemetryReasonDesc(message),
```
Comment (justinmk3) [inline packages/core/src/shared/utilities/messages.ts:96]: Setting `reasonDesc` based on `message`. Maybe in the future, `showMessageWithUrl` should allow the caller to set `id`.

**Review by nkomonen-amazon — APPROVED** [review]

File: packages/core/src/shared/utilities/messages.ts (hunk 1 above)
Comment (justinmk3) [reply packages/core/src/shared/utilities/messages.ts:46]: https://github.com/aws/aws-toolkit-vscode/pull/5776

**Pull request merged**